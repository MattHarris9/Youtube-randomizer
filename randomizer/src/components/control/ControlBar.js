import React from 'react';
import ControlButton from './ControlButton';
import Grid from '@material-ui/core/Grid';
import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';
import PauseCircleOutlineIcon from '@material-ui/icons/PauseCircleOutline';
import RepeatIcon from '@material-ui/icons/Repeat';
import RepeatOneIcon from '@material-ui/icons/RepeatOne';
import ShuffleIcon from '@material-ui/icons/Shuffle';
import SkipNextIcon from '@material-ui/icons/SkipNext';
import SkipPreviousIcon from '@material-ui/icons/SkipPrevious';

export default function ControlBar({
  playNext,
  playPrevious,
  togglePlayPause,
  isPlaying,
  toggleRepeat,
  repeat,
  repeatCurrentSong,
  shuffle,
}) {
  const repeatTooltip = repeatCurrentSong
    ? 'Repeat current song is on'
    : repeat
    ? 'Repeat is on'
    : 'Repeat is off';

  return (
    <Grid container justify="flex-start" alignItems="center" spacing={0}>
      <ControlButton
        tooltip="Previous track"
        icon={<SkipPreviousIcon />}
        click={playPrevious}
      />
      <ControlButton
        tooltip={isPlaying ? 'Pause' : 'Play'}
        active={isPlaying}
        icon={isPlaying ? <PauseCircleOutlineIcon /> : <PlayCircleOutlineIcon />}
        click={togglePlayPause}
      />
      <ControlButton tooltip="Next track" icon={<SkipNextIcon />} click={playNext} />
      <ControlButton
        tooltip="Reshuffle playlist"
        icon={<ShuffleIcon />}
        click={shuffle}
      />
      <ControlButton
        tooltip={repeatTooltip}
        active={repeat || repeatCurrentSong}
        icon={repeatCurrentSong ? <RepeatOneIcon /> : <RepeatIcon />}
        click={toggleRepeat}
      />
    </Grid>
  );
}
