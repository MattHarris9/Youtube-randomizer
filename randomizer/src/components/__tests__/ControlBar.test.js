import React from 'react';
import renderer from 'react-test-renderer';
import ControlBar from '../control/ControlBar';

const mockFn = jest.fn();

describe('Control Bar', () => {
  test('renders empty control bar correctly', () => {
    const tree = renderer.create(<ControlBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders full control bar correctly', () => {
    const props = {
      playNext: mockFn,
      playPrevious: mockFn,
      togglePlayPause: mockFn,
      isPlaying: false,
      toggleRepeat: mockFn,
      repeat: false,
      shuffle: mockFn,
    };
    const tree = renderer.create(<ControlBar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
