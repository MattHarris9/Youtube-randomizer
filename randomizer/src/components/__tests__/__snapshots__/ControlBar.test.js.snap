// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Bar renders empty control bar correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Previous track"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 6h2v12H6zm3.5 6l8.5 6V6z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Play"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Next track"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Reshuffle playlist"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Repeat is off"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Control Bar renders full control bar correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Previous track"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 6h2v12H6zm3.5 6l8.5 6V6z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Play"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Next track"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Reshuffle playlist"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Repeat is off"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
